### PR TITLE
added kustomize edit set label

### PR DESCRIPTION
**Pull request by natasha41575**
#2631

### CODE REVIEW

File: kustomize/internal/commands/edit/set/setlabel_test.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	m, err := kf.Read()
+	if err != nil {
+		t.Errorf("unexpected read error %v", err)
+	}
+	return m
+}
+
+func TestRunSetLabel(t *testing.T) {
+	var o setLabelOptions
+	o.metadata = map[string]string{"owls": "cute", "otters": "adorable"}
```
Comment (monopole) [inline kustomize/internal/commands/edit/set/setlabel_test.go:32]: :)

**Review by monopole — APPROVED** [review]
/lgtm

**Comment by natasha41575** [comment]
@bittner I can't find a section there that explicitly describes kustomize edit commands. The documentation https://kubectl.docs.kubernetes.io/references/kustomize/commonlabels/ seems only to describe how having a commonLabels field in the kustomization file affects the output of kustomize build. For commands the only documentation that we have (to my knowledge) is running `kustomize -h` to see what is available

**Pull request merged**